jdock/src/main/java/io/quarkus/images/modules/GraalVMModule.java

```diff
@@ -88,4 +88,8 @@ public List<Command> commands(BuildContext bc) {
                 new CopyCommand(artifact, "/tmp/" + artifact.name),
                 new RunCommand(script));
     }
+
+    private String removeTrailingZeros(String graalvmVersion) {
+        return graalvmVersion.replaceAll("(\\.0)+$", "");
```

Comment:
The problem with this approach is that if we have both versions `21.0.0` and `21.1.0`, the `21.0.0` will result in a pattern `...21*` which will match the `21.1.0` folder. Maybe the best way would be to run `tar tzf` on the archive to get the actual name of the folder, e.g.:

```
tar tzf ./target/artifacts/graalvm-jdk-21.0.0-linux-x64.tar.gz | head -n 1 | cut -d / -f 1
```

Reply:
Since the extracted folder is just moved to /opt/graalvm I went with: `tar xzf /tmp/archive.tar.gz -C /opt/graalvm --strip-components=1`